### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by adamw-linadm** [cilium/tetragon#2614]
### What happened?

Environment: Kubernetes 1.28.9 on bare metal almalinux 8.9 (each node provisioned  by chef), 22 nodes in cluster,  ~1k pods
Stock config, nothing change, just simple helm install with:
```
helm install tetragon cilium/tetragon -n kube-system
```

 On 10 nodes (22 in cluster) tetragon status is "CrashLoopBackOff" and below is full log output from one crashing pod:
```time="2024-06-28T08:50:09Z" level=info msg="Loaded config from directory" config-dir=/etc/tetragon/tetragon.conf.d/
time="2024-06-28T08:50:09Z" level=info msg="Starting gops server" addr="localhost:8118"
time="2024-06-28T08:50:09Z" level=info msg="Starting tetragon" version=v1.1.2
time="2024-06-28T08:50:09Z" level=info msg="config settings" config="map[bpf-lib:/var/lib/tetragon/ btf: config-dir:/etc/tetragon/tetragon.conf.d/ cpuprofile: data-cache-size:1024 debug:false disable-kprobe-multi:false enable-export-aggregation:false enable-k8s-api:true enable-msg-handling-latency:false enable-pid-set-filter:false enable-pod-info:false enable-policy-filter:true enable-policy-filter-debug:false enable-process-ancestors:true enable-process-cred:false enable-process-ns:false enable-tracing-policy-crd:true event-queue-size:10000 export-aggregation-buffer-size:10000 export-aggregation-window-size:15s export-allowlist:{\"event_set\":[\"PROCESS_EXEC\", \"PROCESS_EXIT\", \"PROCESS_KPROBE\", \"PROCESS_UPROBE\", \"PROCESS_TRACEPOINT\"]} export-denylist:{\"health_check\":true}\n{\"namespace\":[\"\", \"cilium\", \"kube-system\"]} export-file-compress:false export-file-max-backups:5 export-file-max-size-mb:10 export-file-perm:600 export-file-rotation-interval:0s export-filename:/var/run/cilium/tetragon/tetragon.log export-rate-limit:-1 expose-kernel-addresses:false expose-stack-addresses:false field-filters: force-large-progs:false force-small-progs:false generate-docs:false gops-address:localhost:8118 health-server-address::6789 health-server-interval:10 k8s-kubeconfig-path: kernel: kmods:[] log-format:text log-level:info memprofile: metrics-label-filter:namespace,workload,pod,binary metrics-server::2112 netns-dir:/var/run/docker/netns/ pprof-addr: process-cache-size:65536 procfs:/procRoot rb-queue-size:65535 rb-size:0 rb-size-total:0 redaction-filters: release-pinned-bpf:true server-address:localhost:54321 tracing-policy: tracing-policy-dir:/etc/tetragon/tetragon.tp.d username-metadata:disabled verbose:0]"
time="2024-06-28T08:50:09Z" level=info msg="Tetragon pid file creation succeeded" pid=2329951 pidfile=/var/run/tetragon/tetragon.pid
time="2024-06-28T08:50:09Z" level=info msg="BPF detected features: override_return: true, buildid: true, kprobe_multi: false, uprobe_multi false, fmodret: true, fmodret_syscall: true, signal: true, large: true"
time="2024-06-28T08:50:09Z" level=info msg="BPF: successfully released pinned BPF programs and maps" bpf-dir=/sys/fs/bpf/tetragon
time="2024-06-28T08:50:09Z" level=info msg="Enabling policy filtering"
time="2024-06-28T08:50:09Z" level=info msg="BTF discovery: default kernel btf file found" btf-file=/sys/kernel/btf/vmlinux
time="2024-06-28T08:50:09Z" level=info msg="sensor controller waiting on channel"
time="2024-06-28T08:50:09Z" level=info msg="Starting metrics server" addr=":2112"
time="2024-06-28T08:50:09Z" level=info msg="Registering pod delete handler for metrics"
time="2024-06-28T08:50:09Z" level=info msg="Cgroup mode detection succeeded" cgroup.fs=/sys/fs/cgroup cgroup.mode="Unified mode (Cgroupv2)"
time="2024-06-28T08:50:09Z" level=info msg="Supported cgroup controller 'memory' is active on the system" cgroup.controller.hierarchyID=0 cgroup.controller.index=4 cgroup.controller.name=memory cgroup.fs=/sys/fs/cgroup
time="2024-06-28T08:50:09Z" level=info msg="Supported cgroup controller 'pids' is active on the system" cgroup.controller.hierarchyID=0 cgroup.controller.index=11 cgroup.controller.name=pids cgroup.fs=/sys/fs/cgroup
time="2024-06-28T08:50:09Z" level=info msg="Supported cgroup controller 'cpuset' is active on the system" cgroup.controller.hierarchyID=0 cgroup.controller.index=0 cgroup.controller.name=cpuset cgroup.fs=/sys/fs/cgroup
time="2024-06-28T08:50:09Z" level=info msg="Cgroupv2 supported controllers detected successfully" cgroup.controllers="[cpuset cpu io memory hugetlb pids rdma]" cgroup.fs=/sys/fs/cgroup
time="2024-06-28T08:50:09Z" level=info msg="Cgroupv2 controller 'memory' will be used as a fallback for the default hierarchy" cgroup.controller.hierarchyID=0 cgroup.controller.index=4 cgroup.controller.name=memory cgroup.fs=/sys/fs/cgroup
time="2024-06-28T08:50:09Z" level=info msg="Cgroupv2 hierarchy validated successfully" cgroup.fs=/sys/fs/cgroup cgroup.path=/sys/fs/cgroup/init.scope
time="2024-06-28T08:50:09Z" level=info msg="Deployment mode detection succeeded" cgroup.fs=/sys/fs/cgroup deployment.mode=Kubernetes
time="2024-06-28T08:50:09Z" level=info msg="Updated TetragonConf map successfully" NSPID=1 cgroup.controller.hierarchyID=0 cgroup.controller.index=4 cgroup.controller.name=memory cgroup.fs.magic=Cgroupv2 confmap-update=tg_conf_map deployment.mode=Kubernetes log.level=info
time="2024-06-28T08:50:09Z" level=info msg="Enabling Kubernetes API"
time="2024-06-28T08:50:09Z" level=info msg="Waiting for required CRDs" crds="map[tracingpolicies.cilium.io:{} tracingpoliciesnamespaced.cilium.io:{}]"
time="2024-06-28T08:50:09Z" level=info msg="Found CRD" crd=tracingpolicies.cilium.io
time="2024-06-28T08:50:09Z" level=info msg="Found CRD" crd=tracingpoliciesnamespaced.cilium.io
time="2024-06-28T08:50:09Z" level=info msg="Found all the required CRDs"
time="2024-06-28T08:50:09Z" level=info msg="registering policyfilter pod handlers"
time="2024-06-28T08:50:10Z" level=info msg="adding /procRoot/1/root/sys/fs/cgroup/kubepods.slice/kubepods-burstable.slice to cgroup pod directories"
time="2024-06-28T08:50:10Z" level=info msg="adding /procRoot/1/root/sys/fs/cgroup/kubepods.slice/kubepods-besteffort.slice to cgroup pod directories"
time="2024-06-28T08:50:10Z" level=info msg="Initialized informer cache" count=57 informer=pod
time="2024-06-28T08:50:10Z" level=info msg="Disabling Cilium API"
time="2024-06-28T08:50:10Z" level=info msg="Configured redaction filters" redactionFilters=
time="2024-06-28T08:50:10Z" level=info msg="Starting process manager" enableCilium=false enableK8s=true enableProcessCred=false enableProcessNs=false
time="2024-06-28T08:50:10Z" level=info msg="Starting gRPC server" address="localhost:54321" protocol=tcp
time="2024-06-28T08:50:10Z" level=info msg="Configured field filters" fieldFilters="[]"
time="2024-06-28T08:50:10Z" level=info msg="Starting JSON exporter" logger="&{/var/run/cilium/tetragon/tetragon.log 10 0 5 false false -rw------- 0 <nil> {0 0} <nil> {{{} 0} {0 0}}}" request="allow_list:{event_set:PROCESS_EXEC  event_set:PROCESS_EXIT  event_set:PROCESS_KPROBE  event_set:PROCESS_UPROBE  event_set:PROCESS_TRACEPOINT}  deny_list:{health_check:{value:true}}  deny_list:{namespace:\"\"  namespace:\"cilium\"  namespace:\"kube-system\"}"
time="2024-06-28T08:50:10Z" level=info msg="Exporter configuration" enabled=true fileName=/var/run/cilium/tetragon/tetragon.log
time="2024-06-28T08:50:10Z" level=info msg="Starting gRPC health server" address=":6789" interval=10
time="2024-06-28T08:50:10Z" level=info msg="Successfully detected bpftool path" bpftool=/usr/bin/bpftool
time="2024-06-28T08:50:10Z" level=info msg="Successfully detected gops path" gops=/usr/bin/gops
time="2024-06-28T08:50:10Z" level=info msg="BPF: found active BPF resources" bpf-dir=/sys/fs/bpf/tetragon pinned-bpf="[policy_filter_maps tg_conf_map]"
time="2024-06-28T08:50:10Z" level=info msg="Started watching tracing policies"
time="2024-06-28T08:50:10Z" level=info msg="Exit probe on acct_process"
time="2024-06-28T08:50:10Z" level=info msg="BTF file: using metadata file" metadata=/sys/kernel/btf/vmlinux
time="2024-06-28T08:50:10Z" level=info msg="Loading sensor" name=__base__
time="2024-06-28T08:50:10Z" level=info msg="Loading kernel version 4.18.0"
time="2024-06-28T08:50:10Z" level=info msg="tetragon, map loaded." map=execve_map path=/sys/fs/bpf/tetragon/execve_map sensor=__base__
time="2024-06-28T08:50:10Z" level=info msg="tetragon, map loaded." map=tg_execve_joined_info_map path=/sys/fs/bpf/tetragon/tg_execve_joined_info_map sensor=__base__
time="2024-06-28T08:50:10Z" level=info msg="tetragon, map loaded." map=execve_map_stats path=/sys/fs/bpf/tetragon/execve_map_stats sensor=__base__
time="2024-06-28T08:50:10Z" level=info msg="tetragon, map loaded." map=tg_execve_joined_info_map_stats path=/sys/fs/bpf/tetragon/tg_execve_joined_info_map_stats sensor=__base__
time="2024-06-28T08:50:10Z" level=info msg="tetragon, map loaded." map=execve_calls path=/sys/fs/bpf/tetragon/execve_calls sensor=__base__
time="2024-06-28T08:50:10Z" level=info msg="tetragon, map loaded." map=tcpmon_map path=/sys/fs/bpf/tetragon/tcpmon_map sensor=__base__
time="2024-06-28T08:50:10Z" level=info msg="tetragon, map loaded." map=tg_conf_map path=/sys/fs/bpf/tetragon/tg_conf_map sensor=__base__
time="2024-06-28T08:50:10Z" level=info msg="tetragon, map loaded." map=tg_stats_map path=/sys/fs/bpf/tetragon/tg_stats_map sensor=__base__
time="2024-06-28T08:50:10Z" level=info msg="Loading BPF program" Attach=acct_process Program=/var/lib/tetragon/bpf_exit.o Type=kprobe
time="2024-06-28T08:50:10Z" level=info msg="Loading BPF program" Attach=wake_up_new_task Program=/var/lib/tetragon/bpf_fork.o Type=kprobe
time="2024-06-28T08:50:10Z" level=info msg="Loading registered BPF probe" Attach=sched/sched_process_exec Program=/var/lib/tetragon/bpf_execve_event_v53.o Type=execve
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1033810/ns/uts' readlink /procRoot/1033810/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1033810/ns/ipc' readlink /procRoot/1033810/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1033810/ns/mnt' readlink /procRoot/1033810/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1033810/ns/pid_for_children' readlink /procRoot/1033810/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1033810/ns/net' readlink /procRoot/1033810/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1033810/ns/time' readlink /procRoot/1033810/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1033810/ns/time_for_children' readlink /procRoot/1033810/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1033810/ns/cgroup' readlink /procRoot/1033810/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1033828/ns/uts' readlink /procRoot/1033828/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1033828/ns/ipc' readlink /procRoot/1033828/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1033828/ns/mnt' readlink /procRoot/1033828/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1033828/ns/pid_for_children' readlink /procRoot/1033828/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1033828/ns/net' readlink /procRoot/1033828/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1033828/ns/time' readlink /procRoot/1033828/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1033828/ns/time_for_children' readlink /procRoot/1033828/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1033828/ns/cgroup' readlink /procRoot/1033828/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1179874/ns/uts' readlink /procRoot/1179874/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1179874/ns/ipc' readlink /procRoot/1179874/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1179874/ns/mnt' readlink /procRoot/1179874/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1179874/ns/pid_for_children' readlink /procRoot/1179874/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1179874/ns/net' readlink /procRoot/1179874/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1179874/ns/time' readlink /procRoot/1179874/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1179874/ns/time_for_children' readlink /procRoot/1179874/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1179874/ns/cgroup' readlink /procRoot/1179874/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1179892/ns/uts' readlink /procRoot/1179892/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1179892/ns/ipc' readlink /procRoot/1179892/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1179892/ns/mnt' readlink /procRoot/1179892/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1179892/ns/pid_for_children' readlink /procRoot/1179892/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1179892/ns/net' readlink /procRoot/1179892/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1179892/ns/time' readlink /procRoot/1179892/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1179892/ns/time_for_children' readlink /procRoot/1179892/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1179892/ns/cgroup' readlink /procRoot/1179892/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1385173/ns/uts' readlink /procRoot/1385173/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1385173/ns/ipc' readlink /procRoot/1385173/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1385173/ns/mnt' readlink /procRoot/1385173/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1385173/ns/pid_for_children' readlink /procRoot/1385173/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1385173/ns/net' readlink /procRoot/1385173/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1385173/ns/time' readlink /procRoot/1385173/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1385173/ns/time_for_children' readlink /procRoot/1385173/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1385173/ns/cgroup' readlink /procRoot/1385173/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1385192/ns/uts' readlink /procRoot/1385192/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1385192/ns/ipc' readlink /procRoot/1385192/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1385192/ns/mnt' readlink /procRoot/1385192/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1385192/ns/pid_for_children' readlink /procRoot/1385192/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1385192/ns/net' readlink /procRoot/1385192/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1385192/ns/time' readlink /procRoot/1385192/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1385192/ns/time_for_children' readlink /procRoot/1385192/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1385192/ns/cgroup' readlink /procRoot/1385192/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1482551/ns/uts' readlink /procRoot/1482551/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1482551/ns/ipc' readlink /procRoot/1482551/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1482551/ns/mnt' readlink /procRoot/1482551/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1482551/ns/pid_for_children' readlink /procRoot/1482551/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1482551/ns/net' readlink /procRoot/1482551/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1482551/ns/time' readlink /procRoot/1482551/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1482551/ns/time_for_children' readlink /procRoot/1482551/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1482551/ns/cgroup' readlink /procRoot/1482551/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1483717/ns/uts' readlink /procRoot/1483717/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1483717/ns/ipc' readlink /procRoot/1483717/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1483717/ns/mnt' readlink /procRoot/1483717/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1483717/ns/pid_for_children' readlink /procRoot/1483717/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1483717/ns/net' readlink /procRoot/1483717/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1483717/ns/time' readlink /procRoot/1483717/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1483717/ns/time_for_children' readlink /procRoot/1483717/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1483717/ns/cgroup' readlink /procRoot/1483717/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/1536344/ns/uts' readlink /procRoot/1536344/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/1536344/ns/ipc' readlink /procRoot/1536344/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/1536344/ns/mnt' readlink /procRoot/1536344/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/1536344/ns/pid_for_children' readlink /procRoot/1536344/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/1536344/ns/net' readlink /procRoot/1536344/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/1536344/ns/time' readlink /procRoot/1536344/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/1536344/ns/time_for_children' readlink /procRoot/1536344/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/1536344/ns/cgroup' readlink /procRoot/1536344/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/855266/ns/uts' readlink /procRoot/855266/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/855266/ns/ipc' readlink /procRoot/855266/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/855266/ns/mnt' readlink /procRoot/855266/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/855266/ns/pid_for_children' readlink /procRoot/855266/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/855266/ns/net' readlink /procRoot/855266/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/855266/ns/time' readlink /procRoot/855266/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/855266/ns/time_for_children' readlink /procRoot/855266/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/855266/ns/cgroup' readlink /procRoot/855266/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading uts namespace failed" error="namespace '/procRoot/855284/ns/uts' readlink /procRoot/855284/ns/uts: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading ipc namespace failed" error="namespace '/procRoot/855284/ns/ipc' readlink /procRoot/855284/ns/ipc: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading mnt namespace failed" error="namespace '/procRoot/855284/ns/mnt' readlink /procRoot/855284/ns/mnt: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading pid_for_children namespace failed" error="namespace '/procRoot/855284/ns/pid_for_children' readlink /procRoot/855284/ns/pid_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading net namespace failed" error="namespace '/procRoot/855284/ns/net' readlink /procRoot/855284/ns/net: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time namespace failed" error="namespace '/procRoot/855284/ns/time' readlink /procRoot/855284/ns/time: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading time_for_children namespace failed" error="namespace '/procRoot/855284/ns/time_for_children' readlink /procRoot/855284/ns/time_for_children: no such file or directory"
time="2024-06-28T08:50:11Z" level=warning msg="Reading cgroup namespace failed" error="namespace '/procRoot/855284/ns/cgroup' readlink /procRoot/855284/ns/cgroup: no such file or directory"
time="2024-06-28T08:50:11Z" level=info msg="Read ProcFS /procRoot appended 1018/1078 entries"
time="2024-06-28T08:50:11Z" level=warning msg="Failed to find parent for kernel thread 851203"
time="2024-06-28T08:50:11Z" level=info msg="BPF events statistics: 0 received, 0% events loss"
time="2024-06-28T08:50:11Z" level=info msg="Observer events statistics" errors=0 filterDrop=0 filterPass=0 lost=0 received=0
time="2024-06-28T08:50:11Z" level=info msg="sensor controller starts"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x235229d]

goroutine 1 [running]:
github.com/cilium/tetragon/pkg/process.(*Cache).refInc(...)
	/go/src/github.com/cilium/tetragon/pkg/process/cache.go:121
github.com/cilium/tetragon/pkg/process.(*ProcessInternal).RefInc(...)
	/go/src/github.com/cilium/tetragon/pkg/process/process.go:227
github.com/cilium/tetragon/pkg/grpc/exec.(*MsgKThreadInitUnix).HandleMessage(0xc0032c4400)
	/go/src/github.com/cilium/tetragon/pkg/grpc/exec/kthread_init.go:27 +0xbd
github.com/cilium/tetragon/pkg/grpc.(*ProcessManager).Notify(0xc002ee2c60, {0x3000980, 0xc0032c4400})
	/go/src/github.com/cilium/tetragon/pkg/grpc/process_manager.go:59 +0x26
github.com/cilium/tetragon/pkg/observer.(*Observer).observerListeners(0xc00195a960, {0x3000980, 0xc0032c4400})
	/go/src/github.com/cilium/tetragon/pkg/observer/observer.go:54 +0xb0
github.com/cilium/tetragon/pkg/observer.AllListeners(...)
	/go/src/github.com/cilium/tetragon/pkg/observer/observer.go:63
github.com/cilium/tetragon/pkg/sensors/exec/procevents.pushExecveEvents({0x1ba, 0xcfd03, 0x1, 0x484, 0x683c80d8483700, {0xc0061a5a00, 0x73, 0x200}, {0xc005d63980, 0x12, ...}, ...})
	/go/src/github.com/cilium/tetragon/pkg/sensors/exec/procevents/proc_reader.go:226 +0xad8
github.com/cilium/tetragon/pkg/sensors/exec/procevents.pushEvents({0xc003788000, 0x3fb, 0x400})
	/go/src/github.com/cilium/tetragon/pkg/sensors/exec/procevents/proc_reader.go:375 +0x365
github.com/cilium/tetragon/pkg/sensors/exec/procevents.GetRunningProcs()
	/go/src/github.com/cilium/tetragon/pkg/sensors/exec/procevents/proc_reader.go:640 +0xb4
github.com/cilium/tetragon/pkg/sensors/exec.(*execProbe).LoadProbe(0xc0009cc8c0?, {{0xc001174660, 0x14}, 0xc00085f500, 0x41200, 0x0})
	/go/src/github.com/cilium/tetragon/pkg/sensors/exec/exec.go:252 +0x45
github.com/cilium/tetragon/pkg/sensors.loadInstance({0xc001174660, 0x14}, 0xc00085f500, 0x41200, 0x0)
	/go/src/github.com/cilium/tetragon/pkg/sensors/load.go:339 +0x66b
github.com/cilium/tetragon/pkg/sensors.observerLoadInstance({0xc001174660, 0x14}, 0xc00085f500)
	/go/src/github.com/cilium/tetragon/pkg/sensors/load.go:311 +0x63d
github.com/cilium/tetragon/pkg/sensors.(*Sensor).Load(0x4923000, {0xc001174660, 0x14})
	/go/src/github.com/cilium/tetragon/pkg/sensors/load.go:111 +0x69a
main.tetragonExecute()
	/go/src/github.com/cilium/tetragon/cmd/tetragon/main.go:458 +0x19ee
main.execute.func1(0xc0005e3208, {0x2b936ff?, 0x4?, 0x2b93703?})
	/go/src/github.com/cilium/tetragon/cmd/tetragon/main.go:814 +0x16a
github.com/spf13/cobra.(*Command).execute(0xc0005e3208, {0xc000072050, 0x1, 0x1})
	/go/src/github.com/cilium/tetragon/vendor/github.com/spf13/cobra/command.go:987 +0xab1
github.com/spf13/cobra.(*Command).ExecuteC(0xc0005e3208)
	/go/src/github.com/cilium/tetragon/vendor/github.com/spf13/cobra/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/cilium/tetragon/vendor/github.com/spf13/cobra/command.go:1039
main.execute()
	/go/src/github.com/cilium/tetragon/cmd/tetragon/main.go:827 +0x171
main.main()
	/go/src/github.com/cilium/tetragon/cmd/tetragon/tetragon.go:12 +0x13
```

### Tetragon Version

v1.1.2

### Kernel Version

4.18.0-513.24.1.el8_9.x86_64

### Kubernetes Version

1.28.9

**Comment by tpapagian** [comment]
https://github.com/cilium/tetragon/pull/2620 is merged and should solve this issue. It will be part of the next release.

**Comment by adamw-linadm** [comment]
Works fine with fix on 1.2.0. Thanks :)
